pgmpy/inference/dbn_inference.py

```diff
@@ -0,0 +1,98 @@
+from collections import defaultdict
+from pgmpy.inference import VariableElimination
+from pgmpy.factors import TabularCPD
+from pgmpy.inference import Inference
+
+
+class DBNInference(Inference):
+
+    def query(self, variables, evidence=None, elimination_order=None):
+        variable_dict = defaultdict(list)
+        for var in variables:
+            variable_dict[var[1]].append(var[0])
+
+        # This method extracts the max value in time slices
+        timerange = max(variable_dict)
+        if evidence:
+            for node, timeslice in list(evidence):
+                timerange = max(timerange, timeslice)
+
+        int_nodes = self.interface_nodes
+        inference = VariableElimination(self.start_bayesian_model)
+
+        def vect(variables, time):
```

Comment:
And even if it's not being used anywhere else, put it outside the query method until and unless it's very specific to something you are doing in this method.